### PR TITLE
Range-ify std.file remove, getSize

### PR DESCRIPTION
**Pull request by WalterBright**
http://dlang.org/phobos/std_file.html#.remove

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -620,27 +620,70 @@ private void renameImpl(const(char)[] f, const(char)[] t, const(FSChar)* fromz,
...
 Delete file $(D name).
+
+Params:
+    name = string or range of characters representing the file name
+
 Throws: $(D FileException) on error.
  */
-void remove(in char[] name) @trusted
+void remove(R)(R name)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R)
```
Comment (andralex) [inline std/file.d]: Forgot to mention - this is enough: `if (isInputRange!R && isSomeChar!(ElementEncodingType!R))`

File: std/file.d [hunk 2]
```diff
@@ -620,27 +620,70 @@ private void renameImpl(const(char)[] f, const(char)[] t, const(FSChar)* fromz,
...
+            name = namez[0 .. len];
+        }
+        cenforce(core.stdc.stdio.remove(namez) == 0,
             "Failed to remove file " ~ name);
     }
 }
 
-version(Windows) private WIN32_FILE_ATTRIBUTE_DATA getFileAttributesWin(in char[] name) @trusted
+version(Windows) private WIN32_FILE_ATTRIBUTE_DATA getFileAttributesWin(R)(R name)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R)
```
Comment (andralex) [inline std/file.d]: same here, no need for isSomeString

File: std/file.d [hunk 3]
```diff
@@ -655,9 +698,13 @@ version(Windows) private ulong makeUlong(DWORD dwLow, DWORD dwHigh) @safe pure n
...
 Get size of file $(D name) in bytes.
 
+Params:
+    name = string or range of characters representing the file name
+
 Throws: $(D FileException) on error (e.g., file not found).
  */
-ulong getSize(in char[] name) @safe
+ulong getSize(R)(R name)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R)
```
Comment (andralex) [inline std/file.d]: same here, no need for isSomeString

File: std/file.d (hunk 1 above)
Comment (WalterBright) [inline std/file.d]: I was used to strings not being random access ranges!

**Pull request merged**